src/neuroglancer/mesh/backend.ts

```diff
@@ -358,24 +365,33 @@ export class MeshSource extends ChunkSource {
     fragmentSource.meshSource = this;
   }
 
-  getChunk(objectId: Uint64) {
+  getChunk(objectId: Uint64, verifyFragments?: boolean|undefined) {
     const key = getObjectKey(objectId);
     let chunk = <ManifestChunk>this.chunks.get(key);
     if (chunk === undefined) {
+      if (verifyFragments === undefined) verifyFragments = true;
       chunk = this.getNewChunk_(ManifestChunk);
-      chunk.initializeManifestChunk(key, objectId);
+      chunk.initializeManifestChunk(key, objectId, verifyFragments);
       this.addChunk(chunk);
     }
     return chunk;
   }
 
-  getFragmentChunk(manifestChunk: ManifestChunk, fragmentId: FragmentId) {
-    let key = `${manifestChunk.key}/${fragmentId}`;
+  getFragmentChunk(manifestChunk: ManifestChunk, fragmentId: FragmentId, verifyFragment?: boolean|undefined) {
+    let key = fragmentId;
+    if (fragmentId.charAt(0) === '~') {
```

Comment:
I agree with Nico here that this logic is specific to graphene, and in a way it's hidden behavior. I also see your point about trying to avoid far-reaching code changes. No immediate solution comes to mind for me, but Nico's suggestion is definitely worth trying